macros/timeHistoryAnalyzer.py

```diff
@@ -0,0 +1,457 @@
+
+    Each property is stored as a 3D Numpy array with indexes
+    [vfat][strip][time]. The time index -> scan date mapping is exposed in the
+    date attribute.
+
+    Attributes:
+        dates: Array of strings containing the scan dates.
+        mask: Data for the "mask" property,
+        maskReason: Data for the "maskReason" property,
+    """
+
+    def __init__(self, inputDir):
+        """Creates a TimeSeriesData object by reading the files located in the
+        inputDir directory.
+
+        The input directory must contain the following files:
+
+        * gemPlotterOutput_mask_vs_scandate.root
+        * gemPlotterOutput_maskReason_vs_scandate.root
+
+        They are created by plotTimeSeries.py.
+
+        Args:
+            inputDir: The path to the input directory
+        """
+        import ROOT as r
+        from root_numpy import hist2array
+
+        file_mask = r.TFile('%s/gemPlotterOutput_mask_vs_scandate.root' % inputDir, 'READ')
+        file_maskReason = r.TFile('%s/gemPlotterOutput_maskReason_vs_scandate.root' % inputDir, 'READ')
+        file_noise = r.TFile('%s/gemPlotterOutput_noise_vs_scandate.root' % inputDir, 'READ')
+
+        self.mask = [] # [vfat][time][strip]; warning: reordered after loading
+        self.maskReason = [] # [vfat][time][strip]; warning: reordered after loading
+        self.noise = [] # [vfat][time][strip]; warning: reordered after loading
+
+        for vfat in range(0,24):
+            dirname = 'VFAT%d' % vfat
+            dir_mask = file_mask.Get(dirname)
+            dir_maskReason = file_maskReason.Get(dirname)
+            dir_noise = file_noise.Get(dirname)
+
+            hist_mask = dir_mask.Get("h_ROBstr_vs_scandate_Obsmask_VFAT%d" % vfat)
+            hist_maskReason = dir_maskReason.Get("h_ROBstr_vs_scandate_ObsmaskReason_VFAT%d" % vfat)
+            hist_noise = dir_noise.Get("h_ROBstr_vs_scandate_Obsnoise_VFAT%d" % vfat)
+
+            self.mask.append(hist2array(hist_mask))
+            self.maskReason.append(hist2array(hist_maskReason))
+            self.noise.append(hist2array(hist_noise))
+
+            self.dates = [] # [time]
+            for bin in range(hist_mask.GetNbinsX()):
+                self.dates.append(hist_mask.GetXaxis().GetBinLabel(bin + 1))
+                pass
+            pass
+
+        self.dates = np.array(self.dates)
+        self.mask = np.array(self.mask)
+        self.maskReason = np.array(self.maskReason)
+        self.noise = np.array(self.noise)
+
+        self.mask = np.swapaxes(self.mask, 1, 2) # Reorder to [vfat][strip][time]
+        self.maskReason = np.swapaxes(self.maskReason, 1, 2) # Reorder to [vfat][strip][time]
+        self.noise = np.swapaxes(self.noise, 1, 2) # Reorder to [vfat][strip][time]
+
+    def removeBadScans(self, maxMaskedChannelFraction = 0.07):
+        """Finds bad scans and removes them from the data.
+
+        Any scan matching one of the following criteria is considered bad:
+
+        * No channel was masked
```

Comment:
Why is this considered as bad.  A detector with no channel masked is a perfect detector and is what we are striving to build for LS2.  Justify or re-vise.